connectors/slack/template/index.js

```
@@ -1,24 +1,321 @@
+            debugTimeLog('respondToSlack', 'dispatch request to self completed with erroer', error);
+        });
+
+    debugTimeLog('respondToSlack', 'initiated dispatch request to self');
+
+    setTimeout(() => {
+        debugTimeLog('respondToSlack', `responding with HTTP 200 to Slack after ${Date.now() - slackTimestamp}ms from Slack timestamp`);
+        cb(null, { status: 200 });
+    }, 100);
+}
+
+function dispatch(ctx, cb) {
+    debugTime('dispatch', 'starting dispatch');
+    let body = ctx.body;
+
+    // Route all other events
+    if (body && body.api_app_id && body.team_id) {
+        storage.ensureStorage(ctx);
+        storage.ensureCache().then(() => {
+            debugTimeLog('dispatch', 'storage and cache loaded');
+
+            Sdk.debug('Event is for Team ID', body.api_app_id, body.team_id);
+            let handler = storage.teamToHandler[`${body.api_app_id}-${body.team_id}`];
+            if (!handler) {
+                Sdk.debug('No handler registered for this Team ID, not routing');
+                cb();
+                return;
+            }
+
+            Sdk.debug('Handlers registered for this Team ID', handler);
+
+            const promises = handler.map((h) =>
+                Superagent.post(h)
+                    .send(body.event)
+                    .ok((r) => {
+                        debugTimeLog('dispatch', 'handler error return');
+                        if (r.status >= 300) {
+                            Sdk.debug('Handler error', h, r.body);
+                        }
+                        return true;
+                    })
+            );
+
+            debugTimeLog('dispatch', 'initiated dispatch requests to handlers');
+
+            Promise.all(promises)
+                .then(function (result) {
+                    Sdk.debug(`Routed ${handler.length} requests`);
+                })
+                .catch(function (error) {
+                    Sdk.debug('Routing error', error);
```

The function should still return an error - it'll get caught and logged by our internal systems and exposed through statistics.

---

As the code stands today, returning an error here will cause the `Superagent` call in `respondToSlack` to throw thus returning an error to Slack, which we agreed we won't do. We can make this change if we also change that call to swallow all errors. I don't mind, but @tjanczuk can you also chime in?

---

Won't fix